soc/arm/infineon_xmc/4xxx/noinit.ld

```diff
@@ -0,0 +1,11 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2020 Linumiz
+ * Author: Parthiban Nallathambi <parthiban@linumiz.com>
+ *
+ * Extracted from: infineon/XMCLib/devices/XMC4500/Source/GCC/XMC4500x1024.ld
+ */
+
+/* .no_init section contains chipid, SystemCoreClock and trimming data */
+KEEP(*(.no_init))
```

Comment:
is this really needed?

Reply:
Yes, SoC clock initialization uses device specific routines defined [here](https://github.com/linumiz/hal_infineon/blob/master/XMCLib/devices/XMC4500/Source/system_XMC4500.c). SystemCoreClock variable is defined under this special section as [here](https://github.com/linumiz/hal_infineon/blob/master/XMCLib/devices/XMC4500/Source/system_XMC4500.c#L299).

Clock control is currently WIP [here](https://github.com/parthitce/zephyr/commit/fcbb3ca2a34dde97bd04cf7fb16fc8d1f3fa2e39#diff-1f590f20845a91f285809fb66b13b9b4), which will removed dependency with device specific init and probably SystemCoreClock moves into zephyr bss section and calculates the clock value based on Kconfig's.